packages/vscode-extension/src/webview/views/DeviceLocationView.tsx

```diff
@@ -0,0 +1,108 @@
+import { useProject } from "../providers/ProjectProvider";
+import React, { useState, useEffect } from "react";
+
+import "./DeviceLocationView.css";
+import Button from "../components/shared/Button";
+import Label from "../components/shared/Label";
+
+export function DeviceLocationView() {
+  const { project, deviceSettings } = useProject();
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    // @ts-ignore
+    const latitude = event.target[0].value;
+    // @ts-ignore
+    const longitude = event.target[2].value;
+    // @ts-ignore
+    const latDirection = event.target[1].value;
+    // @ts-ignore
+    const lonDirection = event.target[3].value;
+
+    project.updateDeviceSettings({
+      ...deviceSettings,
+      location: {
+        ...deviceSettings.location,
+        latitude: latDirection === "S" ? -parseFloat(latitude) : parseFloat(latitude),
+        longitude: lonDirection === "W" ? -parseFloat(longitude) : parseFloat(longitude),
+      },
+    });
+  };
+
+  const handleDisableLocation = (e: any) => {
+    e.preventDefault();
+
+    let isDisabled;
+    if (e.target.value === "true") {
+      isDisabled = true;
+    } else {
+      isDisabled = false;
+    }
+    project.updateDeviceSettings({
+      ...deviceSettings,
+      location: {
+        ...deviceSettings.location,
+        isDisabled,
+      },
+    });
+  };
+
+  return (
+    <>
+      <form onSubmit={handleSubmit} className="location-controls">
+        <Label>Latitude</Label>
+        <label className="latitude">
+          <div className="picker">
+            <input
+              className="coordinate"
+              type="number"
+              step="0.000001"
+              defaultValue={
+                deviceSettings.location.latitude < 0
+                  ? -deviceSettings.location.latitude
+                  : deviceSettings.location.latitude
+              }
+            />
+            <select
+              className="direction"
+              defaultValue={deviceSettings.location.latitude < 0 ? "S" : "N"}>
```

Comment:
i'm not a fan of magic strings like "S" or "N". They either should be constants on top of the file or grouped as an enum as they never change